### PR TITLE
Fix psalm errors + cleanup + update dev dependencies

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Definitions/ArrayBuilder.php [hunk 1]
```diff
@@ -82,8 +82,7 @@ private function isIntegerIndexed(array $parameters): bool
         $hasStringIndex = false;
         $hasIntegerIndex = false;
 
-        /** @var mixed $parameter */
-        foreach ($parameters as $index => $parameter) {
+        foreach ($parameters as $index => $_parameter) {
```
Comment (rustamwin) [inline src/Definitions/ArrayBuilder.php:85]: Variables prefixed with underscores affect our coding style.

File: src/Definitions/ArrayDefinition.php [hunk 2]
```diff
@@ -106,8 +106,7 @@ public function merge(self $other): self
 
     private function mergeParameters(array $selfParameters, array $otherParameters): array
     {
-        /** @var mixed $param */
-        foreach ($otherParameters as $index => $param) {
+        foreach ($otherParameters as $index => $_param) {
```
Comment (rustamwin) [inline src/Definitions/ArrayDefinition.php:109]: Variables prefixed with underscores affect our coding style.

**Review by rustamwin — CHANGES_REQUESTED** [review]

File: src/Definitions/ArrayBuilder.php (hunk 1 above)
Comment (vjik) [reply src/Definitions/ArrayBuilder.php:85]: This is way for suppress psalm error `UnusedVariable`: https://psalm.dev/docs/running_psalm/issues/UnusedVariable/

File: src/Definitions/ArrayDefinition.php (hunk 2 above)
Comment (vjik) [reply src/Definitions/ArrayDefinition.php:109]: This is way for suppress psalm error `UnusedVariable`: https://psalm.dev/docs/running_psalm/issues/UnusedVariable/

**Pull request merged**